cmd/tuf/app/init.go

```diff
@@ -108,18 +109,30 @@ func InitCmd(ctx context.Context, directory, previous string, targets targetsFla
 				return err
 			}
 		}
-		if err := repo.SetThreshold("root", threshold); err != nil {
+		if err := repo.SetThreshold(role, threshold); err != nil {
 			return err
 		}
 	}
 
-	// Add online snapshot and timestamp keys with a shorter expiration.
+	// Revoke old root keys used for snapshot and timestamp.
+	for _, role := range []string{"snapshot", "timestamp"} {
+		for _, tufKey := range keys {
+			if err := repo.RevokeKey(role, tufKey.IDs()[0]); err != nil {
+				return errors.Wrap(err, "error revoking key")
+			}
+		}
+		if err := repo.SetThreshold(role, 1); err != nil {
+			return err
+		}
+	}
+	// Add online snapshot and timestamp keys with a shorter expiration of three weeks.
 	for role, keyRef := range map[string]string{"snapshot": snapshotRef, "timestamp": timestampRef} {
+		// Add new online keys
 		signerKey, err := pkeys.GetKmsSigningKey(ctx, keyRef)
 		if err != nil {
 			return err
 		}
-		if err := repo.AddVerificationKeyWithExpiration(role, signerKey.Key, time.Now().AddDate(0, 0, 14).UTC()); err != nil {
+		if err := repo.AddVerificationKeyWithExpiration(role, signerKey.Key, time.Now().AddDate(0, 0, 21).UTC()); err != nil {
```

Comment:
Why is 21 hardcoded?

Reply:
3 weeks, `AddDate` only gives params for month, year, and days, so 21 days

Reply:
I see, ok, so hardcoding makes sense. Just doc it pls? 🙂